mw-webapp/src/service/UserService.ts

```diff
@@ -2,37 +2,85 @@ import {
   collection,
   deleteDoc,
   doc,
+  getCountFromServer,
   getDoc,
   getDocs,
+  limit,
+  orderBy,
   query,
+  QueryLimitConstraint,
+  QueryOrderByConstraint,
+  QueryStartAtConstraint,
   setDoc,
+  startAfter,
   updateDoc,
   where,
   WriteBatch,
 } from "firebase/firestore";
 import {db} from "src/firebase";
-import {USER_UUID_FIELD, UserDTO, UserDTOSchema, UserPartialDTOSchema, UsersDTOSchema} from "src/model/DTOModel/UserDTO";
+import {
+  USER_CREATED_AT_FIELD,
+  USER_UUID_FIELD,
+  UserDTO,
+  UserDTOSchema,
+  UserPartialDTOSchema,
+  UsersDTOSchema,
+} from "src/model/DTOModel/UserDTO";
 import {documentSnapshotToDTOConverter} from "src/service/converter/documentSnapshotToDTOConverter";
 import {querySnapshotsToDTOConverter} from "src/service/converter/querySnapshotsToDTOConverter";
 import {querySnapshotToDTOConverter} from "src/service/converter/querySnapshotToDTOConverter";
-import {QUERY_LIMIT} from "src/service/firebaseVariables";
+import {AMOUNT_DOCS_FOR_COUNT_READS, QUERY_LIMIT} from "src/service/firebaseVariables";
 import {RequestOperations} from "src/service/RequestOperations";
 import {getChunksArray} from "src/utils/getChunkArray";
 import {logToConsole} from "src/utils/logToConsole";
 import {PartialWithUuid} from "src/utils/PartialWithUuid";
 
 export const PATH_TO_USERS_COLLECTION = "users";
+const PAGINATION_USERS_AMOUNT = 10;
 
 /**
  * Provides methods to interact with the Users collection in Firestore.
  */
 export class UserService {
 
+  /**
+   * Get UsersDTO amount
+   */
+  public static async getUsersDTOAmount(): Promise<number> {
+    const usersRef = collection(db, PATH_TO_USERS_COLLECTION);
+    const snapshot = await getCountFromServer(usersRef);
+    const usersAmount = snapshot.data().count;
+
+    const readsAmount = Math.ceil(usersAmount / AMOUNT_DOCS_FOR_COUNT_READS);
+
+    logToConsole(`UserService:getUsersDTOAmount: ${readsAmount} ${RequestOperations.READ} operations`);
+
+    return usersAmount;
+  }
+
   /**
    * Get UsersDTO
    */
-  public static async getUsersDTO(): Promise<UserDTO[]> {
-    const usersRaw = await getDocs(collection(db, PATH_TO_USERS_COLLECTION));
+  public static async getUsersDTO(lastUserUuid?: string): Promise<UserDTO[]> {
+    const usersRef = collection(db, PATH_TO_USERS_COLLECTION);
+
+    /**
+     * ExtraRequest
+     */
+    const snapshot = lastUserUuid && await getDoc(doc(db, PATH_TO_USERS_COLLECTION, lastUserUuid));
+    logToConsole(`WayService:getSnapshot: 1 ${RequestOperations.READ} operations`);
+
+    const constraints: (QueryOrderByConstraint | QueryLimitConstraint | QueryStartAtConstraint)[] = [
+      orderBy(USER_CREATED_AT_FIELD, "desc"),
+      limit(PAGINATION_USERS_AMOUNT),
+    ];
+
+    if (snapshot) {
```

Comment:
could youmade it the same way as I made it for filter? (array+spread instead of if+push)